scripts/export_environment.py

```diff
@@ -170,32 +160,45 @@ def resolve_env(env: Environment) -> Environment:
     >>> resolve_env({'x': '{y}', 'y': '42'})
     {'x': '42', 'y': '42'}
 
-    Literal curly braces need to be escaped:
```

Comment:
Where did this case go?

Reply:
This test case was removed because making `resolve_env()` a recursive function to support transitive references removed the ability to support unmatched curly braces as they cause `format_map()` to raise a ValueError. 

That said, I've given this some more thought and come up with a solution that allows for the transitive & generative properties desired while also allowing for unmatched curly braces.